DataStructures/HashMap.java

```diff
@@ -0,0 +1,157 @@
+import java.io.*;
+import java.util.*;
+import java.util.Scanner;
+
+class HashMap {
+
+    // Pair class to store key value pairs
+    static class Pair {
+        int key;
+        int value;
+        Pair next;
+
+        Pair(int key, int value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+
+    int size;
+
+    // Array of pairs to store the key value pairs
+    Pair[] arr;
+
+    // Constructor to initialize the hash map
+    HashMap(int size) {
+        this.size = size;
+        arr = new Pair[size];
+        for (int i = 0; i < size; i++) {
+            arr[i] = null;
+        }
+    }
+
+    // hash function to get the index of the key
+    public int hashFunction(int key) {
+        return key % size;
```

Comment:
Could you please explain the reason of choosing this hash function. What could be its advantages and limitations.

Reply:
This hash function gives fewer collisions or we can say it will repeat the new hash value after (i+size) numbers. 

Advantages:

1. Time complexity of this hash function is less as compared to any other hash function which performs more mathematical operations. 
2. Number of collisions is less. Collision starts when the value is greater than the size of the array.

Limitations:

1. We can get the key by value easily which is not good for security purposes.
2. We can only use integer as input in the hash function, we can't use any other data type.

Note: This is a basic hash function. We can use timestamps or some kind of complex mathematical equation to create a better hash function than this.